src/Models/BaseElement.php

```diff
@@ -712,35 +712,65 @@ public function CMSEditLink($directLink = false)
             return $this->cacheData['cms_edit_link'];
         }
 
+        $link = $this->getElementCMSLink($directLink);
+        $this->extend('updateCMSEditLink', $link);
+
+        if ($link) {
+            $this->cacheData['cms_edit_link'] = $link;
+        }
+
+        return $link;
+    }
+
+    /**
+     * @param bool $directLink
+     * @return null|string
+     */
+    private function getElementCMSLink(bool $directLink)
+    {
         $relationName = $this->getAreaRelationName();
         $page = $this->getPage();
 
+        $link = null;
+
         if (!$page) {
-            $link = null;
-            $this->extend('updateCMSEditLink', $link);
             return $link;
         }
-
-        if (!$page instanceof SiteTree && method_exists($page, 'CMSEditLink')) {
-            $link = Controller::join_links($page->CMSEditLink(), 'ItemEditForm');
-        } else {
+        // If $page is a Page
+        if ($page instanceof SiteTree) {
             $link = $page->CMSEditLink();
+        } else {
+            // Envoke method only if it exists in the $page
+            if (ClassInfo::hasMethod($page, 'CMSEditLink')) {
+                $link = Controller::join_links($page->CMSEditLink(), 'ItemEditForm');
```

Comment:
Is it correct for `ItemEditForm` to be at the end of the url for inline-editable blocks?

Reply:
We don't use 'ItemEditForm' in inline-editable block, we even can return NULL for this block.
But initially was 
```php
        if (!$page instanceof SiteTree && method_exists($page, 'CMSEditLink')) {
            $link = Controller::join_links($page->CMSEditLink(), 'ItemEditForm');
        } else {
            $link = $page->CMSEditLink();
        }
```
I don't want to remove it.

Reply:
Ahh yup, that's fair. I missed that it was already there, so that's fine.